drivers/ondisk_test.go

```diff
@@ -0,0 +1,152 @@
+package drivers
+
+import "testing"
+
+func TestNewOnDiskStorageController(t *testing.T) {
+	t.Parallel()
+	sc := NewOnDiskStorageController("test/controlla")
+	if sc.FSPath != "test/controlla" {
+		t.Fatal("Storage controller created in wrong dir ", sc.FSPath)
+	}
+}
+
+func TestSCConnect(t *testing.T) {
+	t.Parallel()
+	sc := NewOnDiskStorageController("test/controlla")
+
+	err := sc.Connect()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+}
+
+func TestSCDisconnect(t *testing.T) {
+	t.Parallel()
+	sc := NewOnDiskStorageController("test/controlla")
+
+	err := sc.Disconnect()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+}
+
+func TestSCMount(t *testing.T) {
+	t.Parallel()
+	sc := NewOnDiskStorageController("test/controlla")
+	/*
+		mountedPath, err := sc.Mount("")
+
+		if err == nil {
+			t.Fatal("A valid mountedPath name should be provided to storage controller")
+		}
+	*/
```

Comment:
Comments should be removed if unnecessary or uncommented if needed
If it is needed, please leave a note explaining why